distributor/distributor.go

```diff
@@ -391,6 +392,31 @@ func (d *Distributor) LabelValuesForLabelName(ctx context.Context, labelName mod
 	return values, nil
 }
 
+// MetricsForLabelMatchers gets the metrics that match said matchers
```

Comment:
May be good to mention that this may return duplicate metrics, as it doesn't dedupe between ingesters.